### ISSUE TITLE
Guidance on images-next

### ISSUE THREAD

**Issue by timholy** [JuliaIO/QuartzImageIO.jl#28]
If anyone is interested in creating an `images-next` branch, the best model to follow would be my commit to ImageMagick: https://github.com/JuliaIO/ImageMagick.jl/commit/348e9acb929f846a820f4e6e9edfade27db11bc7. From a brief scan of the code in this repo, aside from updating the tests it _seems_ like it could be as simple as:
- returning the plain array and skipping the `Dict` creation, just returning [`buf` directly](https://github.com/JuliaIO/QuartzImageIO.jl/blob/e5dfdd154bcba00404d936af269c31541303c2bd/src/QuartzImageIO.jl#L149-L160).
- some small fixes for [`writemime`](https://github.com/JuliaIO/QuartzImageIO.jl/blob/e5dfdd154bcba00404d936af269c31541303c2bd/src/QuartzImageIO.jl#L290)

**Comment by tlnagy** [comment]
@timholy It looks like there are more problems then just that:

```julia
julia> using TestImages

julia> img = testimage("mandrill")
WARNING: ImageMetadata.Image is deprecated, use ImageMetadata.ImageMeta instead.
  likely near no file:0
WARNING: ImageMetadata.Image is deprecated, use ImageMetadata.ImageMeta instead.
  likely near no file:0
WARNING: ImageMetadata.Image is deprecated, use ImageMetadata.ImageMeta instead.
  likely near no file:0
WARNING: ImageMetadata.Image is deprecated, use ImageMetadata.ImageMeta instead.
  likely near no file:0
WARNING: ImageMetadata.Image is deprecated, use ImageMetadata.ImageMeta instead.
  likely near no file:0
WARNING: ImageMetadata.Image is deprecated, use ImageMetadata.ImageMeta instead.
  likely near no file:0
WARNING: ImageMetadata.Image is deprecated, use ImageMetadata.ImageMeta instead.
  likely near no file:0
WARNING: ImageMetadata.Image is deprecated, use ImageMetadata.ImageMeta instead.
  likely near no file:0
WARNING: ImageMetadata.Image is deprecated, use ImageMetadata.ImageMeta instead.
  likely near no file:0
WARNING: ImageMetadata.Image is deprecated, use ImageMetadata.ImageMeta instead.
  likely near no file:0
WARNING: ImageMetadata.Image is deprecated, use ImageMetadata.ImageMeta instead.
  likely near no file:0
WARNING: ImageMetadata.Image is deprecated, use ImageMetadata.ImageMeta instead.
  likely near no file:0
WARNING: ImageMetadata.Image is deprecated, use ImageMetadata.ImageMeta instead.
  likely near no file:0
WARNING: ImageMetadata.Image is deprecated, use ImageMetadata.ImageMeta instead.
  likely near no file:0
WARNING: ImageMetadata.Image is deprecated, use ImageMetadata.ImageMeta instead.
  likely near no file:0
WARNING: ImageMetadata.Image is deprecated, use ImageMetadata.ImageMeta instead.
  likely near no file:0
WARNING: ImageMetadata.Image is deprecated, use ImageMetadata.ImageMeta instead.
  likely near no file:0
WARNING: ImageMetadata.Image is deprecated, use ImageMetadata.ImageMeta instead.
  likely near no file:0
WARNING: ImageMetadata.Image is deprecated, use ImageMetadata.ImageMeta instead.
  likely near no file:0
WARNING: ImageMetadata.Image is deprecated, use ImageMetadata.ImageMeta instead.
  likely near no file:0
WARNING: ImageMetadata.Image is deprecated, use ImageMetadata.ImageMeta instead.
  likely near no file:0
WARNING: ImageMetadata.Image is deprecated, use ImageMetadata.ImageMeta instead.
  likely near no file:0
in read_and_release_imgsrc at
WARNING: ImageMetadata.Image is deprecated, use ImageMetadata.ImageMeta instead.
  likely near no file:0
WARNING: ImageMetadata.Image is deprecated, use ImageMetadata.ImageMeta instead.
  likely near no file:0
in read_and_release_imgsrc at /Users/tamasnagy/.julia/v0.5/QuartzImageIO/src/QuartzImageIO.jl
WARNING: ImageMetadata.Image is deprecated, use ImageMetadata.ImageMeta instead.
  likely near no file:0
in read_and_release_imgsrc at /Users/tamasnagy/.julia/v0.5/QuartzImageIO/src/QuartzImageIO.jl
WARNING: ImageMetadata.Image is deprecated, use ImageMetadata.ImageMeta instead.
  likely near no file:0
in read_and_release_imgsrc at /Users/tamasnagy/.julia/v0.5/QuartzImageIO/src/QuartzImageIO.jl
Errors encountered while loading "/Users/tamasnagy/.julia/v0.5/TestImages/images/mandrill.tiff".
All errors:
   ErrorException("\"colorspace\" property is ignored, now color is encoded only by the element type.\nSee `colorview` to represent a numeric array as a color array.")
   FileIO.NotInstalledError(:ImageMagick,"")
Fatal error:
ERROR: "colorspace" property is ignored, now color is encoded only by the element type.
See `colorview` to represent a numeric array as a color array.
 in checked_import(::Symbol) at /Users/tamasnagy/.julia/v0.5/FileIO/src/loadsave.jl:12
 in #load#17(::Array{Any,1}, ::Function, ::FileIO.File{FileIO.DataFormat{:TIFF}}) at /Users/tamasnagy/.julia/v0.5/FileIO/src/loadsave.jl:81
 in load(::FileIO.File{FileIO.DataFormat{:TIFF}}) at /Users/tamasnagy/.julia/v0.5/FileIO/src/loadsave.jl:73
 in #load#13(::Array{Any,1}, ::Function, ::String) at /Users/tamasnagy/.julia/v0.5/FileIO/src/loadsave.jl:45
 in testimage(::String) at /Users/tamasnagy/.julia/v0.5/TestImages/src/TestImages.jl:75
```

**Comment by timholy** [comment]
Looks like something must still be calling `Image`. Search for it?

**Comment by tlnagy** [comment]
```
likely near no file:0
```
Any idea on why the error is so useless?

**Comment by timholy** [comment]
Longstanding problem, unfortunately. https://github.com/JuliaLang/julia/issues/16967 is relevant.

**Comment by rsrock** [comment]
As noted over on Images, I have something in the works here.  Loading is fixed, and saving is working for at least one array type (the "autumn_leaves" test).  I'd like to work on something more general for saving-- off to study the ImageMagick code!

**Comment by timholy** [comment]
Great! https://github.com/JuliaImages/Images.jl/blob/master/upgrading_tips.md may be handy, and of course feel free to contribute any tips that you think are "re-usable." I expect a lot of people have private analysis code that may be affected by the recent changes :frowning:, so a collection of "ways to cope" might be helpful.

**Comment by rsrock** [comment]
Closed by 4c19585
